emeval/input/spec_details.py

```diff
@@ -78,11 +78,39 @@ def populate_spec_details(self, curr_spec_entry):
     def fmt(self, ts, fmt_string):
         return arrow.get(ts).to(self.eval_tz).format(fmt_string)
 
-    def get_ground_truth_for_trip(self, trip_id):
-        tl = [t for t in self.curr_spec_entry["data"]["label"]["evaluation_trips"] if t["id"] == trip_id]
-        print(trip_id, len(tl), [t["id"] for t in tl])
+    def get_ground_truth_for_trip(self, trip_id, start_ts, end_ts):
+        tl = [t for t in self.curr_spec_entry["data"]["label"]["evaluation_trips"]
+              if t["id"] == trip_id]
         assert len(tl) == 1
-        return tl[0]
+        tl = tl[0]
+
+        for leg in tl["legs"]:
+            if "route_coords" in leg and isinstance(leg["route_coords"], list):
+                route_coords = [r for r in leg["route_coords"]
+                                if r["properties"]["valid_start_ts"] <= start_ts and r["properties"]["valid_end_ts"] >= end_ts]
+                assert len(route_coords) == 1, f"Not enough route_coords info for {leg['id']} in between timestamps {start_ts} -> {end_ts}"
+                leg["route_coords"] = route_coords[0]
+
+            if "start_loc" in leg and isinstance(leg["start_loc"], list):
+                start_loc = [sl for sl in leg["start_loc"]
+                                if sl["properties"]["valid_start_ts"] <= start_ts and sl["properties"]["valid_end_ts"] >= end_ts]
+                assert len(start_loc) == 1, f"Not enough start_loc info for {leg['id']} in between timestamps {start_ts} -> {end_ts}"
+                leg["start_loc"] = start_loc[0]
+
```

Comment:
seems like we could refactor this into a separate function instead of copy-pasting multiple times.